lib/OpenQA/Config.pm

```diff
@@ -0,0 +1,49 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Config;
+use Mojo::Base -strict, -signatures;
+
+use Config::IniFiles;
+use Exporter qw(import);
+use OpenQA::Log qw(log_info);
+use Mojo::File qw(path);
+
+our @EXPORT = qw(lookup_config_files parse_config_files parse_config_files_as_hash);
+
+sub lookup_config_files ($home, $name, $silent = 0) {
+    my @config_file_paths;
+    for my $path ($ENV{OPENQA_CONFIG}, $home->child('etc', 'openqa'), '/etc/openqa', '/usr/etc/openqa') {
+        next unless defined $path;
+        my $config_path = path($path);
+        my $main_config_file = $config_path->child($name);
+        my $extension = $main_config_file->extname;
+        my $config_name = $main_config_file->basename(".$extension");
+        push @config_file_paths, $main_config_file if -r $main_config_file;
+        push @config_file_paths, @{$config_path->child("$name.d")->list->grep(qr/\.$extension$/)->sort};
```

Comment:
Does this mean that if I do for testing `OPENQA_CONFIG=/my/dir script/openqa` then still settings from /usr/etc/openqa/openqa.ini.d/secret-hidden-file.ini would be effective if not set in /my/dir/openqa.ini?

Reply:
If you do `OPENQA_CONFIG=/my/dir script/openqa` it depends on the contents of `/my/dir`.

If it does not contain any config file (`openqa.ini` or `openqa.ini.d/*.ini`) the directory is ignored and the next one tried (which could be `/usr/etc/…` if none of the other directories contain any config file).

If it contains any config file then the directory is used and none of the other directories (including `/usr/etc/…`) are used at all.